Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm

```diff
@@ -1203,8 +1246,8 @@ static void checkColor(WebCore::CocoaColor *color, std::optional<ColorExpectatio
         RetainPtr attributedText = [NSAttributedString attributedStringWithAttachment:attachment.get()];
 
         [[webView writingToolsDelegate] compositionSession:session.get() didReceiveText:attributedText.get() replacementRange:NSMakeRange(0, 19) inContext:contexts.firstObject finished:YES];
-
-        [webView waitForNextPresentationUpdate];
+        TestWebKitAPI::Util::runFor(0.1_s); // this is needed to let the information load at least a bit or we can fail on a javascript issue.
```

Comment:
```suggestion
        TestWebKitAPI::Util::runFor(0.1_s); // This is needed to let the information load at least a bit or we can fail on a JavaScript issue.
```

Reply:
I'm curious what the JS issue is here? Given `waitForValue` already waits for the image to load.

Reply:
Encountered error: Error Domain=WKErrorDomain Code=4 "A JavaScript exception occurred" UserInfo={WKJavaScriptExceptionLineNumber=1, WKJavaScriptExceptionMessage=TypeError: null is not an object (evaluating 'document.querySelector('img').complete'), WKJavaScriptExceptionColumnNumber=30, WKJavaScriptExceptionSourceURL=file:///..../TestWebKitAPI.resources/, NSLocalizedDescription=A JavaScript exception occurred} while evaluating script: document.querySelector('img').complete

Reply:
Oh, the `<img>` element wasn't created. It might be nicer to update the `waitForValue` to also wait for that to happen.